finish/build.gradle

```diff
@@ -33,7 +33,7 @@ repositories {
 
 // tag::dependencies[]
 dependencies {
-    providedCompile group:'javax.servlet', name:'javax.servlet-api', version:'3.1.0'
+    providedCompile group:'javax.servlet', name:'javax.servlet-api', version:'4.0.0'
```

Comment:
Looks like the latest version is `4.0.1`: https://search.maven.org/search?q=g:javax.servlet%20AND%20a:javax.servlet-api&core=gav

Reply:
yes, should use 4.0.1, see the updated description